util/database.js

```diff
@@ -21,10 +21,15 @@ module.exports = {
 
     query: async (queryText, queryValues) => {
 
-        await pool.connect(() => {
-            console.log('connected');
-        });
+        try {
 
-        return await pool.query(queryText, queryValues);
+            await pool.connect(() => {
```

Comment:
You are not releasing the connection to the pool here. I think this section is why you're having "psql: FATAL: sorry, too many clients already". erroor

Reply:
Thanks Stuart! Fixed the connection leak.